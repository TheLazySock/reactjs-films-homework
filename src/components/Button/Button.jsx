import React from 'react';
import PropTypes from 'prop-types';
import style from './Button.scss';

const Button = (props) => {
  const { title, buttonClass, onClick } = props;
  return (
    <button
      type="button"
      className={style[`${buttonClass}`]}
      onClick={onClick}
    >
      {title}
    </button>
  );
};

Button.defaultProps = {
  onClick: () => { console.log('clicked'); },
};

Button.propTypes = {
  title: PropTypes.string.isRequired,
  buttonClass: PropTypes.string.isRequired,
  onClick: PropTypes.func,
};

export default Button;
