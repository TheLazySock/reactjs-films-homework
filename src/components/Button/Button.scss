.viewInfoBtn {
  padding: 7px 15px;
  border: 1px solid white;
  border-radius: 5px;
  color: white;
  cursor: pointer;
  font-size: 14px;
  background-color: #000;

  &:hover {
    background-color: #313131;
  }

  &:active {
    background-color: #000;
  }
}

.watchVideoBtn {
  padding: 7px 15px;
  margin-right: 20px;
  border: none;
  border-radius: 5px;
  color: white;
  cursor: pointer;
  font-size: 14px;
  background-color: #0aaee4;

  &:hover {
    background-color: lighten(#0aaee4, 15%);
  }

  &:active {
    background-color: #0aaee4;
  }
}
