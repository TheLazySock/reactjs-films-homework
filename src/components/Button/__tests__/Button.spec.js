import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../index';

describe('Button', () => {
  it('renders properly', () => {
    const output = renderer.create(<Button title="btn" buttonClass="cls" />).toJSON();
    expect(output).toMatchSnapshot();
  });

  it('show description on button click', () => {
    const output = renderer.create(<Button title="btn" buttonClass="cls" />);
    output.root.findByType('button').props.onClick();
    expect(output).toMatchSnapshot();
  });
});
