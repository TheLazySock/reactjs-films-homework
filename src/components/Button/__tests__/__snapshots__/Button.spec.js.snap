// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders properly 1`] = `
<button
  className="cls"
  onClick={[Function]}
  type="button"
>
  btn
</button>
`;

exports[`Button show description on button click 1`] = `
<button
  className="cls"
  onClick={[Function]}
  type="button"
>
  btn
</button>
`;
